AlarmClockSpeedTest.cpp

```diff
@@ -0,0 +1,88 @@
+/* 
+ * File:   AlarmClockSpeedTest.cpp
+ * Author: Amanda Carbonari
+ * Created: January 6, 2015 9:00am
+ */
+
+#include <ctime>
+#include <iostream>
+#include <AlarmClock.h> // To test other implementations, change this to #include<header file> or #inlcude<AlarmClock.h.old> for example
+#include <chrono>
+#include <thread>
+using namespace std;
+using namespace std::chrono;
+typedef std::chrono::microseconds microseconds;
+typedef std::chrono::milliseconds milliseconds;
+typedef std::chrono::seconds seconds;
+
+
+template<typename T> void WaitForAlarmClockToExpire(AlarmClock<T>& alerter) {
+   while(!alerter.Expired());
+}
+
+template <typename T> void testReset(unsigned int sleep_time, unsigned int reset_portion=2) {
```

Comment:
Can you add a test that doesn't test the Reset function as well? Just have the original AlarmClock and the new one and let them run for different amounts of time without resetting. I am interested in seeing the overhead and final result of each one. 

The report should have the input sleep time, total time that each took, and overhead time.